### PR TITLE
SNOW-1331876: Support interval in `Window.range_between`

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1331876

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Interval support in RANGE BTEWEEN is just GA last month.

### CODE REVIEW

**Review by sfc-gh-aalam — APPROVED** [review]

File: src/snowflake/snowpark/_internal/analyzer/analyzer.py [hunk 1]
```diff
@@ -722,12 +719,26 @@ def grouping_extractor(
...
+        elif isinstance(boundary, Interval):
+            return window_frame_boundary_expression(boundary.sql, is_following=True)
+        else:
+            offset = self.to_sql_try_avoid_cast(
+                boundary, df_aliased_col_name_to_real_col_name
+            )
+            try:
+                num = int(offset)
+                return window_frame_boundary_expression(str(abs(num)), num >= 0)
+            except Exception:
```
Comment (sfc-gh-jrose) [inline src/snowflake/snowpark/_internal/analyzer/analyzer.py:740]: Can this be more exact?

File: src/snowflake/snowpark/window.py [hunk 2]
```diff
@@ -34,20 +34,29 @@
...
+    end: Union[int, "snowflake.snowpark.Column"],
+) -> Tuple[Expression, Expression]:
+    if isinstance(start, int):
+        if start == 0:
+            boundary_start = CurrentRow()
+        elif start <= Window.UNBOUNDED_PRECEDING:
+            boundary_start = UnboundedPreceding()
+        else:
+            boundary_start = Literal(start)
     else:
```
Comment (sfc-gh-jrose) [inline src/snowflake/snowpark/window.py:48]:
```suggestion
    elif isinstance(start, Column):
```
Might be useful to throw a descriptive error if the typing doesn't match expectation. Similar for end.

**Review by sfc-gh-jrose — COMMENTED** [review]
Just a couple nits.

File: src/snowflake/snowpark/_internal/analyzer/analyzer.py (hunk 1 above)
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/_internal/analyzer/analyzer.py:740]: this is from original code so I prefer not changing it in this PR

File: src/snowflake/snowpark/window.py (hunk 2 above)
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/window.py]: good point, added

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**